src/imagery/i.eodag/i.eodag.py

```diff
@@ -215,54 +264,98 @@ def no_fallback_search(search_parameters, provider):
     # TODO: Would it be useful if user could iterate through
     # the pages manually, and look for the product themselves?
     try:
-        return list(search_result)[0]
+        # Merging the pages into one list with all products
+        return [j for i in search_result for j in i]
     except Exception as e:
         gs.verbose(e)
-        gs.fatal(_("Server error, try again."))
+        gs.fatal(_("Server error, please try again."))
 
 
-def main():
-    # products: https://github.com/CS-SI/eodag/blob/develop/eodag/resources/product_types.yml
+def create_products_dataframe(eo_products):
+    result_dict = {"id": [], "time": [], "cloud_coverage": [], "product_type": []}
+    for product in eo_products:
+        if "id" in product.properties and product.properties["id"] is not None:
+            result_dict["id"].append(product.properties["id"])
+        else:
+            result_dict["id"].append(None)
+        if (
+            "startTimeFromAscendingNode" in product.properties
+            and product.properties["startTimeFromAscendingNode"] is not None
+        ):
+            try:
+                result_dict["time"].append(
+                    normalize_time(product.properties["startTimeFromAscendingNode"])
+                )
+            except:
+                result_dict["time"].append(
+                    product.properties["startTimeFromAscendingNode"]
+                )
+        else:
+            result_dict["time"].append(None)
+        if (
+            "cloudCover" in product.properties
+            and product.properties["cloudCover"] is not None
+        ):
+            result_dict["cloud_coverage"].append(product.properties["cloudCover"])
+        else:
+            result_dict["cloud_coverage"].append(None)
+        if (
+            "productType" in product.properties
+            and product.properties["productType"] is not None
+        ):
+            result_dict["product_type"].append(product.properties["productType"])
+        else:
+            result_dict["product_type"].append(None)
 
-    # setting the envirionmnets variables has to come before the dag initialization
-    setup_environment_variables()
+    df = pd.DataFrame().from_dict(result_dict)
```

Comment:
Pandas is a relatively heavy dependency that is not frequently used in GRASS GIS otherwise. Here pandas features do not really seem to be used. Therefore I would rather avoid pandas and rather print a dict as json or CSV.
I would only use pandas if it provides needed functionality that would be hard to get by other means...

Reply:
I see... I am planning to add a way to sort the results according to a certain parameter e.g. cloud cover, ingestion date, etc... That is why I added pandas DataFrame.

Reply:
@HamedElgizery check sort and order options in i.landsat.download for inspiration, it's simple but maybe useful

Reply:
@ninsbl we suggest to use pandas since also i.sentinel.download is using it (I just checked and it is used only for Google search)